src/libraries/System.Private.CoreLib/src/System/DateOnly.cs

```diff
@@ -28,7 +28,7 @@ namespace System
         // Maps to December 31 year 9999. The value calculated from "new DateTime(9999, 12, 31).Ticks / TimeSpan.TicksPerDay"
         private const int MaxDayNumber = 3_652_058;
 
-        private static int DayNumberFromDateTime(DateTime dt) => (int)(dt.Ticks / TimeSpan.TicksPerDay);
+        private static int DayNumberFromDateTime(DateTime dt) => (int)((ulong)dt.Ticks / TimeSpan.TicksPerDay);
```

Comment:
How does this cast help?

Reply:
It changes the division from signed to unsigned.